packages/rsbuild/src/utils/ast-utils.spec.ts

```diff
@@ -0,0 +1,400 @@
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import {
+  addHtmlTemplatePath,
+  addCopyAssets,
+  addExperimentalSwcPlugin,
+} from './ast-utils';
+
+describe('ast-utils', () => {
+  describe('addHtmlTemplatePath', () => {
+    it('should add the template path to the config when html object does not exist', async () => {
+      // ARRANGE
+      const tree = createTreeWithEmptyWorkspace();
+      tree.write(
+        'apps/my-app/rsbuild.config.ts',
+        `import {defineConfig} from '@rsbuild/core';
+      export default defineConfig({
+      });`
+      );
+
+      // ACT
+      addHtmlTemplatePath(
+        tree,
+        'apps/my-app/rsbuild.config.ts',
+        './src/index.html'
+      );
+
+      // ASSERT
+      expect(tree.read('apps/my-app/rsbuild.config.ts', 'utf-8'))
+        .toMatchInlineSnapshot(`
+        "import {defineConfig} from '@rsbuild/core';
+              export default defineConfig({
+            html: {
+              template: './src/index.html'
+            },
+              });"
+      `);
+    });
+
+    it('should add the template path to the config when html object exists but template is not', async () => {
+      // ARRANGE
+      const tree = createTreeWithEmptyWorkspace();
+      tree.write(
+        'apps/my-app/rsbuild.config.ts',
+        `import {defineConfig} from '@rsbuild/core';
+      export default defineConfig({
+        html: {
+          otherValue: true
+        }
+      });`
+      );
+
+      // ACT
+      addHtmlTemplatePath(
+        tree,
+        'apps/my-app/rsbuild.config.ts',
+        './src/index.html'
+      );
+
+      // ASSERT
+      expect(tree.read('apps/my-app/rsbuild.config.ts', 'utf-8'))
+        .toMatchInlineSnapshot(`
+        "import {defineConfig} from '@rsbuild/core';
+              export default defineConfig({
+                html: {template: './src/index.html',
```

Comment:
Can we fix the formatting here? Moving forward we should assume Prettier may not be used.